src/lambda_function/s3_to_glue/app.py

```diff
@@ -31,38 +73,56 @@ def submit_s3_to_json_workflow(messages: list[dict[str,str]], workflow_name):
     glue_client.put_workflow_run_properties(
         Name=workflow_name,
         RunId=workflow_run["RunId"],
-        RunProperties={"messages": json.dumps(messages)},
+        RunProperties={"messages": json.dumps(objects_info)},
     )
 
 
-def lambda_handler(event, context):
-    """
-    The Lambda entrypoint
-
-    Given a list of files in s3 bucket to process, submit the keys and bucket
-    of the new files to a S3 to JSON Glue workflow.
+def lambda_handler(event, context) -> None:
+    """This main lambda function will be triggered by a
+        SQS event and will poll the SQS queue for
+        all available S3 event messages
 
     Args:
-        event (dict): A fileview object
-        context (obj) : An object that provides methods and properties that
-        provides info about the invocation, function, and execution environment
-
-    Returns:
-        (None) Submits new file records to a Glue workflow.
+        event (json): SQS event
+        context: NA
     """
-    messages = []
-    for record in event["Records"]:
-        bucket_name = record["s3"]["bucket"]["name"]
-        object_key = record["s3"]["object"]["key"]
-        message = {"source_bucket": bucket_name, "source_key": object_key}
-        if "owner.txt" not in object_key:
-            messages.append(message)
-    if len(messages) > 0:
-        logger.info(
+    # Initialize SQS client
+    sqs = boto3.client("sqs")
+    s3_objects_info = []
+    if "Records" in event:
+        for record in event["Records"]:
+            s3_event_records = json.loads(record["body"])
+            if "Records" in s3_event_records:
+                for s3_event in s3_event_records["Records"]:
+                    bucket_name = s3_event["s3"]["bucket"]["name"]
+                    object_key = s3_event["s3"]["object"]["key"]
```

Comment:
I guess my question is I don't know, but since we should only be allowing S3 events into the SQS queue due to our bucket configuration, it seems they will always be supposedly guaranteed to have this structure. I also don't know if the S3 event will always follow this structure of being under `body` and will contain `Records`. I guess the question here is do we prefer to have a exception to the expected structure run into the `KeyError` or only log it? If we just log it, we may not be as likely to catch it if everything shows up as passed (and it just gets deleted from lambda since it gets processed "successfully")

Reply:
If anything other than an SQS/S3 event is sent to our lambda that would be a fluke. Meaning: impossible without a human unintentionally submitting a malformed event to the lambda. I think it makes sense to assume that the event will be well formatted (that's for AWS to test and be concerned about, after all). If somehow a message with another format makes its way to our lambda (during testing), it makes sense for the lambda to raise an exception. The message would eventually end up in the DLQ where we can purge it. This scenario occurring in production just isn't feasible unless AWS is broken and we don't need to be prepared for that.

Reply:
Since we are going with "assume well formatted", then I don't think we need to test that these keys will always be defined in the event since they are supposed to be a part of the S3 events we get.